src/go/rpk/pkg/cli/cmd/redpanda/mode_test.go

```diff
@@ -29,7 +29,7 @@ import (
 func fillRpkConfig(path, mode string) *config.Config {
 	conf := config.Default()
 	val := mode == config.ModeProd
-	conf.Redpanda.DeveloperMode = !val
```

Comment:
`developer_mode` ended up as a node config property, so you can undo the part of this that removed it from the node config object